core_lib_generator/config_collectors/data_access.py

```diff
@@ -4,52 +4,53 @@
 from core_lib.helpers.string import any_to_pascal
 
 
-def generate_data_access_template(db_entities: dict) -> dict:
-    data_access = {}
+def generate_data_access_template(db_entities: list) -> list:
+    data_access = []
 
     def is_exists(user_input: str):
-        return False if user_input in data_access else True
+        for da in data_access:
+            if da['key'] == user_input:
+                return False
+        return True
 
-    for db_conn in db_entities:
-        for entity in db_entities[db_conn]:
-            if entity == 'migrate':
-                continue
+    if db_entities:
+        for entity in db_entities:
+            entity_name = entity['key']
+            db_conn = entity['db_connection']
             default = (
-                f'{any_to_pascal(entity)}DataAccess'
-                if f'{any_to_pascal(entity)}DataAccess' not in data_access
+                f'{any_to_pascal(entity_name)}DataAccess'
+                if is_exists(f'{any_to_pascal(entity_name)}DataAccess')
                 else None
             )
             data_access_name = any_to_pascal(
                 input_str(
-                    f'What is the name of the data access? (Database connection: {db_conn}, Entity: {entity})',
+                    f'What is the name of the data access? (Database connection: {db_conn}, Entity: {entity_name})',
                     default,
                     False,
                     is_exists,
                 )
             )
             is_crud_soft_delete_token = False
             is_crud_soft_delete = False
-            if db_entities[db_conn][entity]['is_soft_delete'] and db_entities[db_conn][entity]['is_soft_delete_token']:
+            if entity['is_soft_delete'] and entity['is_soft_delete_token']:

```

Comment:
entity.get('is_soft_delete')./ .get('is_soft_delete_token') is much more safe and will not raise any exceptions